src/system/Notice/Notice.tsx

```diff
@@ -5,43 +5,61 @@
  */
 import React from 'react';
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import { MdError, MdWarning, MdCheckCircle, MdInfo } from 'react-icons/md';
 
 /**
  * Internal dependencies
  */
 import { Box, Flex, Heading, Card } from '../';
+import { ThemeUIStyleObject } from 'theme-ui';
 
-const colorSystemVariant = color =>
-	( { warning: 'warning', error: 'error', alert: 'error', success: 'success', info: 'info' }[
-		color
-	] );
+interface NoticeIconProps {
+	color: string;
+	variant: keyof COLOR_VARIANTS;
+}
```

Comment:
this can simplified with literal string union type

```suggestion
type ColorVariant = 'alert' | 'error' | 'info' | 'success' | 'warning';

interface NoticeIconProps {
	color: string;
	variant: ColorVariant;
}
```